Libraries/Image/RCTImageLoader.m

```diff
@@ -377,9 +371,26 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
                                                    userInfo:nil], nil);
           return;
         }
-      }
+        
+        NSString *responseDate = ((NSHTTPURLResponse *)response).allHeaderFields[@"Date"];
+        NSString *cacheKey = RCTCacheKeyForImage(imageURLRequest.URL.absoluteString, size, scale, resizeMode, responseDate);
+
+        UIImage *image = [_decodedImageCache objectForKey:cacheKey];
+        if (image) {
+          completionBlock(nil, image);
+          return;
+        }
 
-      // Call handler
+        [weakSelf decodeImageData:data
```

Comment:
Can you nest this in the `else` block and have a shared return statement after the block?